config/v1/types.go

```diff
@@ -80,6 +81,123 @@ type ConfigMapReference struct {
 	Key string `json:"filename,omitempty"`
 }
 
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Build holds cluster-wide information on how to handle builds. The canonical name is `cluster`
+type Build struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec holds user-settable values for the build controller configuration
+	// +optional
+	Spec BuildSpec `json:"spec,omitempty"`
+}
+
+type BuildSpec struct {
+	// AdditionalTrustedCA is a reference to a ConfigMap containing additional CAs that
+	// should be trusted for image pushes and pulls during builds.
+	// +optional
+	AdditionalTrustedCA ConfigMapReference `json:"additionalTrustedCA,omitempty"`
+	// BuildDefaults controls the default information for Builds
+	// +optional
+	BuildDefaults *BuildDefaults `json:"buildDefaults,omitempty"`
+	// BuildOverrides controls override settings for builds
+	// +optional
+	BuildOverrides *BuildOverrides `json:"buildOverrides,omitempty"`
+}
+
+type BuildDefaults struct {
+	// GitHTTPProxy is the location of the HTTPProxy for Git source
+	// +optional
+	GitHTTPProxy string `json:"gitHTTPProxy,omitempty"`
+
+	// GitHTTPSProxy is the location of the HTTPSProxy for Git source
+	// +optional
+	GitHTTPSProxy string `json:"gitHTTPSProxy,omitempty"`
+
+	// GitNoProxy is the list of domains for which the proxy should not be used
+	// +optional
+	GitNoProxy string `json:"gitNoProxy,omitempty"`
+
+	// Env is a set of default environment variables that will be applied to the
+	// build if the specified variables do not exist on the build
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// SourceStrategyDefaults are default values that apply to builds using the
+	// source strategy.
+	// +optional
+	SourceStrategyDefaults *SourceStrategyDefaults `json:"sourceStrategyDefaults,omitempty"`
+
+	// ImageLabels is a list of docker labels that are applied to the resulting image.
+	// User can override a default label by providing a label with the same name in their
+	// Build/BuildConfig.
+	// +optional
+	ImageLabels []ImageLabel `json:"imageLabels,omitempty"`
+
+	// NodeSelector is a selector which must be true for the build pod to fit on a node
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
```

Comment:
This stands out to me.  When would a cluster admin want this as a default, but not an override?  

This feels familiar.  @bparees did we talk about this when it landed the first time?

Reply:
as i recall the distinction had to do with whether we set the key or not, if it's already set in the build itself.

defaulter:  if the user set the key, we don't touch it.
overrider: we stomp the key

Reply:
(i'm comfortable with making the case that only override is needed and we get rid of the default path, for now)

Reply:
> (i'm comfortable with making the case that only override is needed and we get rid of the default path, for now)

I think we should do this.